### PR TITLE
Liner Window/Door Sensor

### PR DESCRIPTION
**Pull request by SkynetHome**
My attempt at adding Linear Product WADWAZ-1 to Z-wave binding!

### CODE REVIEW

**Comment by cdjackson** [comment]
Thanks.
This looks ok, but two things to do...

1) Format the file - click on Source | Format in Eclipse - this just makes it look neater.
2) You need to update the [product file](https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database#product-file). HABmin should tell you the ID/TYPE values, so you just need to add the new device with these values, and reference the XML file...

**Comment by SkynetHome** [comment]
Ok I will do that now!

**Comment by SkynetHome** [comment]
Ok i just edited the original because I found errors in source code. Formatted that and updated Products file and re-committed. Do I need to create a new pull request for that commit?

**Comment by SkynetHome** [comment]
Never mind I see that this pull was updated!

**Comment by cdjackson** [comment]
Sorry - the addition to the products file is wrong. As there is already a manufacturer for Linear, you need to find the existing entry (around line 30) and add the new product into here...

**Comment by SkynetHome** [comment]
Ok, sorry about that. Fixing it now!

**Comment by cdjackson** [comment]
No probs - I’m updating the wiki each time you make a mistake :)

**Comment by SkynetHome** [comment]
LOL...glad I can be of service. If you need someone to NOOB proof anything I am your guy!

File: bundles/binding/org.openhab.binding.zwave/database/linear/wadwaz-1.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			<id>0xFF</id>
+		</Class>
+	</CommandClasses>
+	<Configuration>
+		<Parameter>
+			<Index>1</Index>
+			<Type>short</Type>
+			<Default>0</Default>
+			<Size>2</Size>
+			<label Lang="en">IN1 Alarm Cancellation Delay alarm Cancellation
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/linear/wadwaz-1.xml]: Sorry - these should be capital 'L' - ie ```<Label>``` and ```</Label>```. Same comment a few lines down.

**Comment by SkynetHome** [comment]
On it!

**Comment by cdjackson** [comment]
Thanks. It looks ok, but for some reason it's showing a conflict above (ie the we can't merge this error). I'll try and resolve this - I might close this PR and create a new one if I need to...

**Comment by SkynetHome** [comment]
Sorry for the trouble. It could be something to do with the command classes. I noticed they were different than other examples I looked at. I tried to compare with Fibaro Window/Door Sensor but it has more functions like temp sensing. These are just really basic contact sensors. Let me know if there is anything else I can do. As a side note I have noticed that I have some GE Wall switches that work but show up in habmin as GE and then an id and type number instead of with a product name and description. There is no DB file existing for these specific switches, but they work. Will I break something by creating a product entry for them?

**Comment by cdjackson** [comment]
No problem. It wasn’t anything to do with what you did. The issue is that someone else also changed this file while you were adding your stuff. Since they changed the same part of the file (adding another Linear device into the product file at the same place you did) this caused a merge error that had to be manually resolved… It was just easier for me to pull your changes across to my repo and fix it :)

You should periodically make sure you pull the latest changes from the openhab repo into your repo - generally doing this before you start changing the file will likely reduce the chances of this happening…


> As a side note I have noticed that I have some GE Wall switches that work but show up in habmin as GE and then an id and type number instead of with a product name and description. There is no DB file existing for these specific switches, but they work.
Yes - the database isn’t actually needed unless a) you want to name the device (which is always nice), or b) you need to add configuration. Since some devices don’t have configuration (i.e. no configuration or association section in the file) it doesn’t matter if they are in the DB or not.


> Will I break something by creating a product entry for them?
Not at all - feel free :)
As above, if the device has no config, you don’t need to add the device file - just add the product into the products.xml file, but don’t add the config file link.

Thanks
Chris

**Comment by SkynetHome** [comment]
Excellent. I am afraid this little experience is going to cost me some money as now I want to go look for more to add to the growing DB. I am enjoying learning to code. If there are ever any little projects like this that you dont want to mess with I would be happy to work on them for you! Just let me know how I can help!

**Comment by SkynetHome** [comment]
Two more questions:

1. When I "sync" with Github Windows is that "pulling the latest openhab repo" into my repo?
2. When I import the Github clone into Eclipse IDE most everything has red x's most of which say some particular line in the code cannot be resolved. Is this normal or do I need to install a library of some kind?

**Comment by cdjackson** [comment]
> On 26 Jun 2015, at 20:51, Mike Van Dusen <notifications@github.com> wrote:
> 
> Just let me know how I can help!

Thanks - I’ll keep that in mind :)

**Comment by cdjackson** [comment]
> When I "sync" with Github Windows is that "pulling the latest openhab repo" into my repo?
Yes  - depending on the software, it’s normally pulling the openhab ‘master’ branch into your local branch.

> When I import the Github clone into Eclipse IDE most everything has red x's most of which say some particular line in the code cannot be resolved. Is this normal or do I need to install a library of some kind?
It’s not normal. When you installed the repo/IDE, did you run through the steps to run a bunch of scripts to generate the models?

One other point since I noticed your PR was from your master… Typically, it’s not best practice to code directly in your repos master branch. What I do is when I want to make a new feature, I change to my local master, pull the openhab master to my local repo so it’s up to date, then create a local branch which has a name of the new feature. Then all changes for that feature are in that branch rather than master, and you keep master ‘clean’ - just effectively a copy of what is in the main openhab master.

**Comment by SkynetHome** [comment]
No I did not run any scripts. I am sorry, I am a very fast learner but this is all new to me. I have done a little bit of Python Coding but that is all. Before I started working on this today I didnt even have the Java JDK, it took me a few minutes just to get Ecplise IDE to run as I had to edit the .ini file to direct it to my javaw file. I will google it and see if I cant figure out how to get repo/IDE properly installed!

**Comment by SkynetHome** [comment]
Per usual I jumped in without reading the entire wiki. Installing per the IDE Setup instructions now. Hopefully this will prevent future problems!

**Comment by SkynetHome** [comment]
Still struggling to get all the errors and exceptions out of Eclipse. I think it has something to do with Maven 3, but in the meantime is there something I need to do to realize the affects of the device update I made? My contacts are connecting to zstick configuring and healing but are not actually updating their state when opened or closed. Furthermore none of the information I added is appearing next to the devices in habmin. Still comes up as Linear devices with just ID and Type Number. No name no configuration options!

**Comment by cdjackson** [comment]
> Still struggling to get all the errors and exceptions out of Eclipse.
> 
I sympathise - it’s not the simplest thing :(

> I think it has something to do with Maven 3,
> 
I don’t think Maven should impact things if you’re only compiling through Eclipse (but I might be wrong). 

> but in the meantime is there something I need to do to realize the affects of the device update I made?
> 
Yes, you will need to update the binding. Normally this would have been auto built, but the build system was playing up and I think it might have been disabled. 

You have 2 / 3 options -:
1) Wait until tomorrow - openhab will be rebuilt later tonight and you can grab the latest from Cloudbees
2) Compile it in Eclipse if you can get it to work. If Eclipse is compiling ok (bit I think it’s not?) then you can right click on the zwave binding and click Export. Select Plug-in development and then the … and fragments option.
3) Compile using Maven if you can get that to work (go to the openhab folder and run “mvn clean install”

**Comment by SkynetHome** [comment]
I am running Eclipse on my Windows 7 pc and openhab is running on Raspberry Pi 2!  Dont I need to update the binding on the raspberry pi?

**Comment by cdjackson** [comment]
Yes, but you will need to compile the binding on the PC, then copy it into the addons folder on the RPi.

**Comment by SkynetHome** [comment]
OIC...is it a problem that I am running JDK 1.8 after following everything in IDE Setup wiki I have 935 errors and just over 1000 warnings stuff like (AbstractItemsJavaValidator - Cannot be resolved)

**Comment by cdjackson** [comment]
> On 26 Jun 2015, at 23:55, Mike Van Dusen <notifications@github.com> wrote:
> 
> OIC...is it a problem that I am running JDK 1.8
> 
I don’t think so, but I’m not 100% sure. I know others are using 1.8 so should be ok (I use 1.7)

> after following everything in IDE Setup wiki I have 935 errors and just over 1000 warnings stuff like (AbstractItemsJavaValidator - Cannot be resolved)
> 
Not sure what that is - sorry. Have you tried using maven - from a dos prompt, go to the openhab folder (i.e. the git repository main folder) and run "mvn clean install" and see what happens.

**Comment by SkynetHome** [comment]
I will try that but I just noticed that I do not have any errors in zwave binding...so here is an embarrassing question...how do I compile it? I have auto build checked so shouldnt it be built already. What do I need to copy and add to my addons folder?

**Comment by cdjackson** [comment]
If you right mouse click on the zwave binding in the “Project Explorer” and click the Export menu. Select "Plug-in development" and then the “Deployable plugins and fragments" option.  This will (hopefully!!) put the binding into a “plugins” folder which is in the openhab repository root folder.

**Comment by SkynetHome** [comment]
Excellent will try that in a sec, I just started the dos command you referenced above and its doing all kinds of stuff. When that finishes I will reopen Eclipse and see whats doing.  I can thank you enough for all your help. I apologize for being so annoying with questions all day!

**Comment by cdjackson** [comment]
No probs. 

Just to note that the mvn command is really separate to Eclipse, so it probably won’t change things there… However, if it works, then you will find the JAR in the folder openhab/bundles/bindings/org.openhab.binding.zwave/target (from memory!).  Also, if this works, then once you’ve run the whole thing once (as you are doing now), then if you change to the openhab/bundles/bindings/org.openhab.binding.zwave folder and run “mvn clean install” it will compile just the zwave binding which will be a LOT quicker than compiling everything…

Good luck :)

**Comment by SkynetHome** [comment]
Just tried to export while dos is still running got to building then threw an error log with 65 errors

**Comment by cdjackson** [comment]
Ok - if maven is still running that’s a good sign as it will stop as soon as it gets an error…  It builds incrementally, so you might find the zwave binding turns up in the target folder at some stage before it completes.

**Comment by SkynetHome** [comment]
clean install failed as well here are just some of the items skipped.
error

**Comment by cdjackson** [comment]
I don’t see zwave in the list - I guess there’s no chance it was in the successful part of the log?

**Comment by SkynetHome** [comment]
sure wish I hadnt closed that log I guess I could test again on just zwave. But what do you mean zwave will show up in the target folder

**Comment by cdjackson** [comment]
>  But what do you mean zwave will show up in the target folder

The output file (i.e. the zwave JAR file) will be in the folder openhab/bundles/bindings/org.openhab.binding.zwave/target - it will be worth looking in there to see if it’s built...

**Comment by SkynetHome** [comment]
here is the error log when I run mvn clean install on binding.zwave
zwaveerror

**Comment by cdjackson** [comment]
Hmmm - sorry - I’m not sure what’s going on...

**Comment by SkynetHome** [comment]
You know i wonder if the problem is that I am not running 1.8 I have only 1.7 bindings

**Comment by cdjackson** [comment]
> On 27 Jun 2015, at 00:40, Mike Van Dusen <notifications@github.com> wrote:
> 
> You know i wonder if the problem is that I am not running 1.8 I have only 1.7 bindings
> 
Probably not - generally speaking the version doesn’t matter (I assume you’re talking about the version of OH, and not Java).

**Comment by SkynetHome** [comment]
yes

**Comment by SkynetHome** [comment]
I dont suppose the fact that I am running 64bit eclipse is the cause of my problems is it?

**Comment by cdjackson** [comment]
> 
> I dont suppose the fact that I am running 64bit eclipse is the cause of my problems is it?
> 
No - it shouldn’t matter.

Where did you get Eclipse from? Did you use the Yoxos installer, or did you download from the Eclipse website?

**Comment by SkynetHome** [comment]
I downloaded from Eclipse. I have tried a ton of stuff to get it to work
right. Downloaded different Eclipse bundles from Eclipse. Added Maven and
tried running various openhab things with errors as maven clean. Followed
all instructions of IDE setup except the one about running the the four
scripts because I dont get an option to "Genereate abc Build" maybe thats
the problem but I cant figure out what is causing that either. Most all of
the errors are "unable to resolve org.openhab.(binding).SNAPSHOT-1.8.0. The
warnings all seem to pertain to miscalled items in the various java script
due to missing library as far as I can tell. To top it off, I pulled down
the cloudbees binding lastnight with my zwave addition in it. My devices
are recognized and test shown as being alive and done in openhab but wont
hold configurations and are not transmitting OPEN/CLOSED states to openhab.
So I guess its back to the drawing board there as well! Hopefully today
will be a more sucessful day!
On Jun 27, 2015 2:56 AM, "Chris Jackson" <notifications@github.com> wrote:
>
> >
> > I dont suppose the fact that I am running 64bit eclipse is the cause of
my problems is it?
> >
> No - it shouldn’t matter.
>
> Where did you get Eclipse from? Did you use the Yoxos installer, or did
you download from the Eclipse website?

**Comment by cdjackson** [comment]
> On 27 Jun 2015, at 17:04, Mike Van Dusen <notifications@github.com> wrote:
> 
> I downloaded from Eclipse.

I’d suggest using the Yoxos installer for openhab (see here https://github.com/openhab/openhab/wiki/IDE-Setup#yoxos-instructions <https://github.com/openhab/openhab/wiki/IDE-Setup#yoxos-instructions>). It should make sure that you download all the relevant bits


> the errors are "unable to resolve org.openhab.(binding).SNAPSHOT-1.8.0. 

This is probably because the openhab runtime isn’t configured - again, if you use the installer it will sort this out for you.



>  but wont
> hold configurations

Probably the configuration isn’t being transferred from the binding to the device. The configuration will not be transferred until the device wakes up. Do you have “zwave:masterController=true” in the zwave config file? This will help. I’d suggest setting this, deleting the XML files in the openhab/etc/zwave folder, and restarting the binding/openhab. Then wake up the device so it configured. Then when you change the configuration, you should wake the device up again. In theory it will wake up by itself, but it might take an hour or two to transfer the configuration to the device.

**Comment by SkynetHome** [comment]
Habmin2 shows possible associations habmin does not. The device wakes and
tries to heal gets to  updateroute and then this either does nothing or I
rewake the device and it fails and deletes the route. In this process it
loses changes I made to configuration so it never associates with node 1
which I assume is why it doesnt talk to z stick. The other weird thing is
that I cannot set individual group associations. Like on my Fibaro I can
choose to only associate group 3 with the controller but on the device I
added if I associate one group with node 3 it automatically associates all
3. If I try to remove one association it removes all. I think I just need
to fix the binding. I built it using the editor instead of just writing the
source java code. I will have a look at the code and see if I can get it
straightened out and issue another pull request.

**Comment by cdjackson** [comment]
Did the binding create an XML file for this node in /etc/zwave? If so, please post it here.

Can you provide a link to the manual for this device, or the pepper1 database?

Why do you think it has 3 association groups - is this from the manual? I've not been able to find the info anywhere

**Comment by SkynetHome** [comment]
Sure I will provide manual in one sec. That's just it I don't think it has
3 association groups. I was copying from a similar device when building my
binding and just rushed it. Give me a sec and I'll upload the manual.

**Comment by SkynetHome** [comment]
Here is a link to the manual. http://www.nortekcontrol.com/pdf/manuals/WADWAZ_install.pdf

**Comment by cdjackson** [comment]
Yep - it looks like it only supports a single group, however, it also doesn't mention the configuration class at all, so are you sure that the two config parameters should be there?

**Comment by SkynetHome** [comment]
Here are the node xml 

<node>
  <deviceClass>
    <basicDeviceClass>ROUTING_SLAVE</basicDeviceClass>
    <genericDeviceClass>BINARY_SENSOR</genericDeviceClass>
    <specificDeviceClass>ROUTING_SENSOR_BINARY</specificDeviceClass>
  </deviceClass>
  <homeId>0x18531b5</homeId>
  <nodeId>13</nodeId>
  <version>4</version>
  <manufacturer>0x14f</manufacturer>
  <deviceId>0x102</deviceId>
  <deviceType>0x2001</deviceType>
  <listening>false</listening>
  <frequentlyListening>false</frequentlyListening>
  <routing>true</routing>
  <healState>GETASSOCIATIONS @ 2015-06-27T17:56:51+0000</healState>
  <supportedCommandClasses>
    <entry>
      <commandClass>WAKE_UP</commandClass>
      <WakeUpCommandClass>
        <version>2</version>
        <instances>1</instances>
        <targetNodeId>1</targetNodeId>
        <interval>3600</interval>
        <minInterval>600</minInterval>
        <maxInterval>604800</maxInterval>
        <defaultInterval>3600</defaultInterval>
        <intervalStep>200</intervalStep>
        <lastWakeup>2015-06-27 17:56:51.488 UTC</lastWakeup>
      </WakeUpCommandClass>
    </entry>
    <entry>
      <commandClass>NO_OPERATION</commandClass>
      <noOperationCommandClass>
        <version>1</version>
        <instances>1</instances>
      </noOperationCommandClass>
    </entry>
    <entry>
      <commandClass>ALARM</commandClass>
      <alarmCommandClass>
        <version>1</version>
        <instances>1</instances>
        <alarms/>
        <isGetSupported>true</isGetSupported>
      </alarmCommandClass>
    </entry>
    <entry>
      <commandClass>MANUFACTURER_SPECIFIC</commandClass>
      <manufacturerSpecificCommandClass>
        <version>1</version>
        <instances>1</instances>
      </manufacturerSpecificCommandClass>
    </entry>
    <entry>
      <commandClass>BASIC</commandClass>
      <basicCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </basicCommandClass>
    </entry>
    <entry>
      <commandClass>ASSOCIATION</commandClass>
      <associationCommandClass>
        <version>1</version>
        <instances>1</instances>
        <configAssociations>
          <entry>
            <int>1</int>
            <associationGroup>
              <Index>1</Index>
              <Nodes/>
            </associationGroup>
          </entry>
        </configAssociations>
        <maxGroups>1</maxGroups>
      </associationCommandClass>
    </entry>
    <entry>
      <commandClass>VERSION</commandClass>
      <versionCommandClass>
        <version>1</version>
        <instances>1</instances>
        <libraryType>LIB_CONTROLLER_BRIDGE</libraryType>
        <protocolVersion>3.67</protocolVersion>
        <applicationVersion>4.84</applicationVersion>
      </versionCommandClass>
    </entry>
    <entry>
      <commandClass>BATTERY</commandClass>
      <batteryCommandClass>
        <version>1</version>
        <instances>1</instances>
        <batteryLevel>100</batteryLevel>
        <batteryLow>false</batteryLow>
        <isGetSupported>true</isGetSupported>
      </batteryCommandClass>
    </entry>
    <entry>
      <commandClass>SENSOR_BINARY</commandClass>
      <binarySensorCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </binarySensorCommandClass>
    </entry>
  </supportedCommandClasses>
  <nodeNeighbors>
    <int>1</int>
    <int>2</int>
    <int>4</int>
    <int>6</int>
    <int>8</int>
    <int>9</int>
    <int>10</int>
    <int>11</int>
  </nodeNeighbors>
  <lastSent>2015-06-27 17:56:51.488 UTC</lastSent>
  <lastReceived>2015-06-27 17:56:51.522 UTC</lastReceived>
  <applicationUpdateReceived>true</applicationUpdateReceived>
</node>

**Comment by cdjackson** [comment]
Thanks. This also indicates just one association group and no configuration parameters...

**Comment by SkynetHome** [comment]
<node>
  <deviceClass>
    <basicDeviceClass>ROUTING_SLAVE</basicDeviceClass>
    <genericDeviceClass>BINARY_SENSOR</genericDeviceClass>
    <specificDeviceClass>ROUTING_SENSOR_BINARY</specificDeviceClass>
  </deviceClass>
  <homeId>0x18531b5</homeId>
  <nodeId>14</nodeId>
  <version>4</version>
  <manufacturer>0x14f</manufacturer>
  <deviceId>0x102</deviceId>
  <deviceType>0x2001</deviceType>
  <listening>false</listening>
  <frequentlyListening>false</frequentlyListening>
  <routing>true</routing>
  <healState>DONE @ 2015-06-27T08:26:23+0000</healState>
  <supportedCommandClasses>
    <entry>
      <commandClass>WAKE_UP</commandClass>
      <WakeUpCommandClass>
        <version>2</version>
        <instances>1</instances>
        <targetNodeId>1</targetNodeId>
        <interval>3600</interval>
        <minInterval>600</minInterval>
        <maxInterval>604800</maxInterval>
        <defaultInterval>3600</defaultInterval>
        <intervalStep>200</intervalStep>
        <lastWakeup>2015-06-27 08:26:23.58 UTC</lastWakeup>
      </WakeUpCommandClass>
    </entry>
    <entry>
      <commandClass>NO_OPERATION</commandClass>
      <noOperationCommandClass>
        <version>1</version>
        <instances>1</instances>
      </noOperationCommandClass>
    </entry>
    <entry>
      <commandClass>ALARM</commandClass>
      <alarmCommandClass>
        <version>1</version>
        <instances>1</instances>
        <alarms/>
        <isGetSupported>true</isGetSupported>
      </alarmCommandClass>
    </entry>
    <entry>
      <commandClass>MANUFACTURER_SPECIFIC</commandClass>
      <manufacturerSpecificCommandClass>
        <version>1</version>
        <instances>1</instances>
      </manufacturerSpecificCommandClass>
    </entry>
    <entry>
      <commandClass>BASIC</commandClass>
      <basicCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </basicCommandClass>
    </entry>
    <entry>
      <commandClass>ASSOCIATION</commandClass>
      <associationCommandClass>
        <version>1</version>
        <instances>1</instances>
        <configAssociations>
          <entry>
            <int>1</int>
            <associationGroup>
              <Index>1</Index>
              <Nodes/>
            </associationGroup>
          </entry>
        </configAssociations>
        <maxGroups>1</maxGroups>
      </associationCommandClass>
    </entry>
    <entry>
      <commandClass>VERSION</commandClass>
      <versionCommandClass>
        <version>1</version>
        <instances>1</instances>
        <libraryType>LIB_CONTROLLER_BRIDGE</libraryType>
        <protocolVersion>3.67</protocolVersion>
        <applicationVersion>4.84</applicationVersion>
      </versionCommandClass>
    </entry>
    <entry>
      <commandClass>BATTERY</commandClass>
      <batteryCommandClass>
        <version>1</version>
        <instances>1</instances>
        <batteryLevel>100</batteryLevel>
        <batteryLow>false</batteryLow>
        <isGetSupported>true</isGetSupported>
      </batteryCommandClass>
    </entry>
    <entry>
      <commandClass>SENSOR_BINARY</commandClass>
      <binarySensorCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </binarySensorCommandClass>
    </entry>
  </supportedCommandClasses>
  <nodeNeighbors>
    <int>1</int>
    <int>2</int>
    <int>4</int>
    <int>6</int>
    <int>7</int>
    <int>8</int>
    <int>9</int>
    <int>10</int>
    <int>11</int>
  </nodeNeighbors>
  <lastSent>2015-06-27 07:26:03.214 UTC</lastSent>
  <lastReceived>2015-06-27 08:26:23.58 UTC</lastReceived>
  <applicationUpdateReceived>true</applicationUpdateReceived>
</node>

**Comment by SkynetHome** [comment]
CdJackson: "Yep - it looks like it only supports a single group, however, it also doesn't mention the configuration class at all, so are you sure that the two config parameters should be there?"

No in fact I deleted the configuration part of my binding when I first wrote it and then added it back before I submitted to match the Fibaro example. Definitely think that was a mistake!

**Comment by SkynetHome** [comment]
I just edited the xml in eclipse and committed but not sure where it went and it wont let me sync says I dont have permission.

**Comment by cdjackson** [comment]
What branch are you editing. You need to make sure it's your repository, and you're not inadvertently trying to do this directly into the openhab repo?

**Comment by SkynetHome** [comment]
thats what it was was...I have made a mess of this Eclipse trying to get rid of all these errors and warnings

**Comment by SkynetHome** [comment]
I just deleted everything. I had workspaces all over the place and multiple clones. Going to clean install and see if I can get this right.

**Comment by SkynetHome** [comment]
Going the Yoxos route this time installing now. Hopefully this will workout better. Thanks for continuing to bear with me! I really appreciate your patience!

**Comment by cdjackson** [comment]
No worries - good luck :)

**Comment by SkynetHome** [comment]
OMG...I think I got it. The YOXOS route was definitely the way to go. Got through the IDE setup and all errors are gone. Still have warnings but no errors.

**Comment by SkynetHome** [comment]
Committed and synced change to binding!

**Comment by cdjackson** [comment]
Cool - and don’t worry about the warnings…

Are you making another PR for the changes then? Or maybe now that you can compile the binding you can test it before doing the PR (although I think it will be ok with no config and only 1 association).

**Comment by SkynetHome** [comment]
Yeah just realized I probably should have made pr before committing. Trying
to fix that now.

**Comment by cdjackson** [comment]
No - you always need to commit before doing the PR. Commit, push to your repo, then go to Github website, go to the openhab repo (either yours or the main one) and there should be a label across the top of the page asking if you want to make a PR (this stays there fore a few hours after you push to Githib)

**Comment by SkynetHome** [comment]
Thanks I was drowning at that point, almost created a random pull that
wasn't attached to my push.

**Comment by SkynetHome** [comment]
How can I issue a separate pull for the new Switches I comitted? I think it got added to my other open pull request. Can you not have more than one pull open at a time?

**Comment by cdjackson** [comment]
You can have as many as you like - you just need to put them into a separate branch. If you put them into the same branch, then they will be part of that PR as soon as you push it to the server.

**Comment by SkynetHome** [comment]
Should I re-comit in new branch? Also tried to build the binding now that
errors are gone but no joy :(

**Comment by SkynetHome** [comment]
Sweet binding compiled I needed to check the option  "Use class files compiled in the workspace"

**Pull request closed without merging**